src/Host/Monitor/Impl/BrokerManager.cs

```diff
@@ -47,29 +50,27 @@ public class BrokerManager {
                     psi.WorkingDirectory = assemblyRoot;
 
                     if (Properties.Settings.Default.UseDifferentBrokerUser) {
-                        await CredentialManager.SetCredentialsOnProcess(psi);
+                        await CredentialManager.SetCredentialsOnProcessAsync(psi);
                     }
                     
                     _brokerProcess = new Process() { StartInfo = psi };
-
+                    _brokerProcess.EnableRaisingEvents = true;
+                    _brokerProcess.Exited += ProcessExited;
                     _brokerProcess.Start();
                 }
 
-                _brokerProcess.EnableRaisingEvents = true;
-                MainWindow.SetStatusText($"Broker Process running ... {_brokerProcess.Id}");
-                _brokerProcess.Exited += _brokerProcess_Exited;
                 AutoRestart = true;
+                MainWindow.SetStatusText($"Broker Process running ... {_brokerProcess.Id}");
             });
         }
 
         public static void ResetAutoStart() {
             _autoRestartCount = 0;
         }
 
-
-        private static async void _brokerProcess_Exited(object sender, EventArgs e) {
-            if (AutoRestart && ++_autoRestartCount <= AutoRestartCount) {
-                await StartBrokerInstanceAsync();
+        private static void ProcessExited(object sender, EventArgs e) {
+            if (AutoRestart && ++_autoRestartCount <= AutoRestartMaxCount) {
+                CreateOrAttachToBrokerInstanceAsync().DoNotWait();
```

Comment:
I'm not sure which thread this event handler will run on, but probably it's a threadpool thread, so expected exceptions need to be suppressed accordingly.